modules/oeFax/messageUI.php

```diff
@@ -0,0 +1,486 @@
+        Dropzone.autoDiscover = false;
+        $(function () {
+            fileTypes = '';
+            if (Service === '2') {
+                fileTypes = 'application/pdf';
+            }
+            var faxQueue = new Dropzone("#faxQueue", {
+                paramName: 'fax',
+                url: 'faxProcessUploads',
+                dictDefaultMessage: queueMsg,
+                clickable: true,
+                enqueueForUpload: true,
+                maxFilesize: 25,
+                acceptedFiles: fileTypes,
+                uploadMultiple: false,
+                addRemoveLinks: true,
+                init: function (e) {
+                    let ofile = '';
+                    this.on("addedfile", function (file) {
+                        console.log('new file added ', file);
+                        ofile = file;
+                    });
+                    this.on("sending", function (file) {
+                        console.log('upload started ', file);
+                        $('.meter').show();
+                    });
+                    this.on("success", function (file, response) {
+                        let thisFile = response;
+                        console.log('upload success ', thisFile);
+                        sendFax(thisFile, 'queue');
+                    });
+                    this.on("queuecomplete", function (progress) {
+                        $('.meter').delay(999).slideUp(999);
+                    });
+                    this.on("removedfile", function (file) {
+                        console.log(file);
+                    });
+                }
+            });
+        });
+        $(function () {
+            $('.datepicker').datetimepicker({
+                <?php
+                $datetimepicker_timepicker = false;
+                $datetimepicker_showseconds = false;
+                $datetimepicker_formatInput = false;
+                require($GLOBALS['srcdir'] . '/js/xl/jquery-datetimepicker-2-5-4.js.php');
+                ?>
+            });
+            var dateRange = new Date(new Date().setDate(new Date().getDate() - 7));
+            $("#fromdate").val(dateRange.toJSON().slice(0, 10));
+            $("#todate").val(new Date().toJSON().slice(0, 10));
+            if (Service === '2') {
+                $(".ringcentral").hide();
+            }
+            // populate
+            retrieveMsgs();
+        });
+
+        var wait = '<span id="wait"><?php echo xlt("Fetching Remote") . '..';?><i class="fa fa-cog fa-spin fa-2x"></i></span>';
+
+        var sendFax = function (filePath, from = '') {
+            let btnClose = '<?php echo xlt("Cancel"); ?>';
+            let title = '<?php echo xlt("Send To Contact"); ?>';
+            let url = top.webroot_url + '/modules/oeFax/contact.php?isDocuments=false&isQueue=' +
+                encodeURIComponent(from) + '&file=' + filePath; // do not encode filePath
```

Comment:
I was gonna tell you to encode filePath, but then saw the comment :)

Reply:
I know how you love new escape functions. need one that will normalize filepaths first.